### ISSUE TITLE
Unmounted `/boot/efi` causes all sources to be deleted

### ISSUE THREAD

**Issue by gootorov** [projg2/eclean-kernel#54]
I ran `eclean-kernel -n 3` with unmounted `/boot/efi` and it resulted in all kernel sources to be deleted:
```
Preserving currently running kernel (6.10.10-gentoo-x86_64)
Legend:
[-] file being removed
[x] file does not exist (anymore)
[+] file being kept (used by other kernels)

* Removing kernel modules-only 6.10.5-gentoo-x86_64 (vmlinuz does not exist)
 [-] /usr/src/linux-6.10.5-gentoo
 [-] /lib/modules/6.10.5-gentoo-x86_64
* Removing kernel modules-only 6.11.0-gentoo-x86_64 (vmlinuz does not exist)
 [-] /usr/src/linux-6.11.0-gentoo
 [-] /lib/modules/6.11.0-gentoo-x86_64
Removed 2 kernels
```

My suggestion - `eclean-kernel` should probably check for mounted `/boot/efi`.

Possibly related to https://github.com/projg2/eclean-kernel/issues/19

**Comment by mgorny** [comment]
Do you have `pymountboot` installed? What kernels does it `--list`?

**Comment by gootorov** [comment]
It is installed:
```
❯ eix pymountboot
* dev-python/pymountboot
     Available versions:  0.2.3-r2^t
     IUSE (all versions): debug test PYTHON_TARGETS="python3_10 python3_11 python3_12 python3_13"
     Installed versions:  Version:   0.2.3-r2^t
                          Date:      07:59:59 PM 07/23/2024
                          USE:       -debug -test PYTHON_TARGETS="python3_12 -python3_10 -python3_11 -python3_13"
                          DEPEND:    >=sys-apps/util-linux-2.20 dev-lang/python:3.12
                          RDEPEND:   ${DEPEND} >=sys-libs/glibc-2.39-r9
                          BDEPEND:   virtual/pkgconfig dev-lang/python:3.12 >=dev-python/gpep517-15[python_targets_python3_12(-)] >=dev-python/setuptools-69.0.3[python_targets_python3_12(-)]
                          EAPI:      8
     Best versions/slot:  0.2.3-r2^t
     Homepage:            https://github.com/projg2/pymountboot/
     Description:         Python extension module to (re)mount /boot
     License:             BSD
```
Though I'm not sure how to `--list`, sorry. I don't see any `pymountboot` binaries, and as a python module, I get the following:
```
❯ python -m pymountboot --list
/usr/lib/python-exec/python3.12/python: No code object available for pymountboot
```

Here's `--list-kernels --debug` from `eclean-kernel` though. Also, a correction - my EFI partition is mounted to `/boot/efi`, not `/boot`. Hopefully that's not significant.

When `/boot/efi` is mounted:
```
DEBUG:root:Sorter: <ecleankernel.sort.VersionSort object at 0x7f5725edeff0>
DEBUG:root:Layout: <ecleankernel.layout.blspec.BlSpecLayout object at 0x7f5725c2f110>
DEBUG:root:Bootloader failed: <class 'ecleankernel.bootloader.lilo.LILO'>
DEBUG:root:Bootloader failed: <class 'ecleankernel.bootloader.grub2.GRUB2'>
DEBUG:root:Bootloader failed: <class 'ecleankernel.bootloader.grub.GRUB'>
DEBUG:root:Bootloader failed: <class 'ecleankernel.bootloader.yaboot.Yaboot'>
DEBUG:root:Bootloader: <ecleankernel.bootloader.symlinks.Symlinks object at 0x7f5725c2f170>
bls 6.11.0-gentoo-x86_64 [6.11.0-gentoo-x86_64]
- emptydir: /boot/efi/machine_id/6.11.0-gentoo-x86_64
- initramfs: /boot/efi/machine_id/6.11.0-gentoo-x86_64/initrd
- vmlinuz: /boot/efi/machine_id/6.11.0-gentoo-x86_64/linux
- misc: /boot/efi/machine_id/6.11.0-gentoo-x86_64/microcode-amd
- modules: /lib/modules/6.11.0-gentoo-x86_64
- build: /usr/src/linux-6.11.0-gentoo
- last modified: 2024-09-28 19:46:50
bls 6.10.10-gentoo-x86_64 [6.10.10-gentoo-x86_64]
- emptydir: /boot/efi/machine_id/6.10.10-gentoo-x86_64
- initramfs: /boot/efi/machine_id/6.10.10-gentoo-x86_64/initrd
- vmlinuz: /boot/efi/machine_id/6.10.10-gentoo-x86_64/linux
- misc: /boot/efi/machine_id/6.10.10-gentoo-x86_64/microcode-amd
- modules: /lib/modules/6.10.10-gentoo-x86_64
- build: /usr/src/linux-6.10.10-gentoo
- last modified: 2024-09-14 15:38:22
bls 6.10.5-gentoo-x86_64 [6.10.5-gentoo-x86_64]
- emptydir: /boot/efi/machine_id/6.10.5-gentoo-x86_64
- initramfs: /boot/efi/machine_id/6.10.5-gentoo-x86_64/initrd
- vmlinuz: /boot/efi/machine_id/6.10.5-gentoo-x86_64/linux
- misc: /boot/efi/machine_id/6.10.5-gentoo-x86_64/microcode-amd
- last modified: 2024-08-17 09:46:28
```

When `/boot/efi` isn't mounted:
```
DEBUG:root:Sorter: <ecleankernel.sort.VersionSort object at 0x7f62950deff0>
DEBUG:root:Layout failed: <class 'ecleankernel.layout.blspec.BlSpecLayout'>; exception: /boot/[EFI/]loader not found
DEBUG:root:Layout: <ecleankernel.layout.std.StdLayout object at 0x7f62950fa000>
DEBUG:root:Bootloader failed: <class 'ecleankernel.bootloader.lilo.LILO'>
DEBUG:root:Bootloader failed: <class 'ecleankernel.bootloader.grub2.GRUB2'>
DEBUG:root:Bootloader failed: <class 'ecleankernel.bootloader.grub.GRUB'>
DEBUG:root:Bootloader failed: <class 'ecleankernel.bootloader.yaboot.Yaboot'>
DEBUG:root:Bootloader: <ecleankernel.bootloader.symlinks.Symlinks object at 0x7f6294e3f3b0>
other 6.11.0-gentoo-x86_64 [None]
- modules: /lib/modules/6.11.0-gentoo-x86_64
- build: /usr/src/linux-6.11.0-gentoo
- last modified: 2024-09-28 19:46:50
other 6.10.10-gentoo-x86_64 [None]
- modules: /lib/modules/6.10.10-gentoo-x86_64
- build: /usr/src/linux-6.10.10-gentoo
- last modified: 2024-09-26 08:29:48
```

The bootloader is `systemd-boot`

**Comment by mgorny** [comment]
Weird. It should refuse to proceed since there are no vmlinuz files.

**Comment by gootorov** [comment]
Ah! I think I've figured out what went wrong the first time.

So, I had a `/boot/MACHINE_ID` and a `/boot/efi/MACHINE_ID` directory at the same time. The former one was some mistake/leftover from the initial installation. I've removed it and now `eclean-kernel -n 3` raises an error that no kernels are found, without deleting any sources/modules, as expected.

Sorry for raising the alarms, and thanks for the quick responses.
